rxsample/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/contract/UserDetailsContract.java

```diff
@@ -0,0 +1,63 @@
+package com.mateuszkoslacz.moviper.rxsample.viper.contract;
+
+import android.widget.ImageView;
+
+import com.hannesdorfmann.mosby.mvp.MvpPresenter;
+import com.hannesdorfmann.mosby.mvp.lce.MvpLceView;
+import com.mateuszkoslacz.moviper.iface.interactor.MoviperRxInteractor;
+import com.mateuszkoslacz.moviper.iface.routing.MoviperViewHelperRxRouting;
+import com.mateuszkoslacz.moviper.iface.viewhelper.MoviperViewHelper;
+import com.mateuszkoslacz.moviper.rxsample.data.model.User;
+import com.mateuszkoslacz.moviper.rxsample.viper.view.activity.UserDetailsActivity;
+
+import rx.Observable;
+
+public interface UserDetailsContract {
+
+    interface Presenter extends MvpPresenter<View> {
+        // Defines what methods the View can invoke on the Presenter.
+        // In most cases there will be user interactions and View lifecycle events.
+
+        void onViewCreated();
+
+        void onAvatarClicked(String avatarUrl);
+    }
+
+    interface View extends MvpLceView<User> {
+        // Defines what methods the Presenter can invoke on the View
+        // In most cases there will be manipulating ui and displaying data or errors.
+        // In Super Rx version it also provides getters for Observables emmiting user click events.
+
+        void bindDataToViews(User user);
+
+        void setLoginAndAvatarForUser(User user);
+    }
+
+    interface Interactor extends MoviperRxInteractor {
+        // Defines what methods the Presenter can invoke on the Interactor.
+        // In most cases there will be data saving and querying.
+        // It's just a marker interface.
+
+        Observable<User> getUserForUsername(String user);
+    }
+
+    interface Routing extends MoviperViewHelperRxRouting<ViewHelper> {
+        // Defines what methods the Presenter can invoke on the Routing.
+        // In most cases there will be starting another activities, services and using system
+        // framework, ie. scheduling alarms or sending broadcasts.
+        // In the case of a fragment being the view, there also will be manipulating
+        // the root Activity, ie. switching fragments.
+
+        void startFullscreenPhotoActivity(String avatarUrl);
+
+        User getUserDataIntent(UserDetailsActivity userDetailsActivity);
+    }
+
+    interface ViewHelper extends MoviperViewHelper {
+        // Defines what Android views the Routing can get from the Viper View.
+        // There should only be Android View getter methods to provide the Routing elements
+        // to be used on advanced Android transitions.
+
+        ImageView getAvatarImageView();
```

Comment:
Please use pure Views in ViewHelper to avoid encouraging user to do anything else than transitions.